src/context/mapStateSlice/index.ts

```diff
@@ -41,10 +42,32 @@ export const mapStateSlice = createSlice({
   name: 'mapState',
   initialState,
   reducers: {
-    addLayer: ({ layers, ...rest }, { payload }: PayloadAction<LayerType>) => ({
-      ...rest,
-      layers: layers.filter(layer => keepLayer(layer, payload)).concat(payload),
-    }),
+    addLayer: ({ layers, ...rest }, { payload }: PayloadAction<LayerType>) => {
+      // Check if a layer belongs to a group.
+      if (!payload.group) {
+        return {
+          ...rest,
+          layers: layers
+            .filter(layer => keepLayer(layer, payload))
+            .concat(payload),
+        };
+      }
+
+      const { name } = payload.group;
+      // Get all layers that belong to a group.
+      const items = Object.values(LayerDefinitions).filter(l => {
+        if (l.group?.name === name) {
+          return true;
+        }
+
+        return false;
```

Comment:
I meant that you can go even one step further and just:

`Object.values(LayerDefinitions).filter(l => (l.group?.name === name))`